doc/rados/operations/add-or-rm-osds.rst

```diff
@@ -2,195 +2,214 @@
 
    .. prompt:: bash $
 
       ssh {new-osd-host}
       ceph-osd -i {osd-num} --mkfs --mkkey
 
-   The directory must be empty before you can run ``ceph-osd``.
+   Make sure that the directory is empty before running ``ceph-osd``.
 
-#. Register the OSD authentication key. The value of ``ceph`` for
-   ``ceph-{osd-num}`` in the path is the ``$cluster-$id``.  If your
-   cluster name differs from ``ceph``, use your cluster name instead:
+#. Register the OSD authentication key by running a command of the following
+   form:
 
    .. prompt:: bash $
 
       ceph auth add osd.{osd-num} osd 'allow *' mon 'allow rwx' -i /var/lib/ceph/osd/ceph-{osd-num}/keyring
 
-#. Add the OSD to the CRUSH map so that the OSD can begin receiving data. The
-   ``ceph osd crush add`` command allows you to add OSDs to the CRUSH hierarchy
-   wherever you wish. If you specify at least one bucket, the command
-   will place the OSD into the most specific bucket you specify, *and* it will
-   move that bucket underneath any other buckets you specify. **Important:** If
-   you specify only the root bucket, the command will attach the OSD directly
-   to the root, but CRUSH rules expect OSDs to be inside of hosts.
-
-   Execute the following:
+   This presentation of the command has ``ceph-{osd-num}`` in the listed path
+   because many clusters have the name ``ceph``. However, if your cluster name
+   is not ``ceph``, then the string ``ceph`` in ``ceph-{osd-num}`` needs to be
+   replaced with your cluster name. For example, if your cluster name is
+   ``cluster1``, then the path in the command should be
+   ``/var/lib/ceph/osd/cluster1-{osd-num}/keyring``.
+
+#. Add the OSD to the CRUSH map by running the following command. This allows
+   the OSD to begin receiving data. The ``ceph osd crush add`` command can add
+   OSDs to the CRUSH hierarchy wherever you wish. If you specify one or more
+   buckets, the command places the OSD in the most specific of those buckets,
+   and it moves that bucket underneath any other buckets that you have
+   specified. **Important:** If you specify only the root bucket, the command
+   will attach the OSD directly to the root, but CRUSH rules expect OSDs to be
+   inside of hosts.
 
    .. prompt:: bash $
 
       ceph osd crush add {id-or-name} {weight}  [{bucket-type}={bucket-name} ...]
 
-   You may also decompile the CRUSH map, add the OSD to the device list, add the
-   host as a bucket (if it's not already in the CRUSH map), add the device as an
-   item in the host, assign it a weight, recompile it and set it. See
-   `Add/Move an OSD`_ for details.
+   Note that there is another way to add a new OSD to the CRUSH map: decompile
+   the CRUSH map, add the OSD to the device list, add the host as a bucket (if
+   it is not already in the CRUSH map), add the device as an item in the host,
+   assign the device a weight, recompile the CRUSH map, and set the CRUSH map.
+   For details, see `Add/Move an OSD`_.
```

Comment:
With recent Ceph releases this is rarely necessary

Reply:
Accepted.